mkautodoc/extension.py

```diff
@@ -42,28 +45,67 @@ def get_params(signature: inspect.Signature) -> typing.List[str]:
     render_kw_only_separator = True
 
     for parameter in signature.parameters.values():
-        value = parameter.name
-        if parameter.default is not parameter.empty:
-            value = f"{value}={parameter.default!r}"
+        param = etree.Element("span", {"class": "autodoc-param-definition"})
 
+        value = parameter.name
         if parameter.kind is parameter.VAR_POSITIONAL:
             render_kw_only_separator = False
             value = f"*{value}"
         elif parameter.kind is parameter.VAR_KEYWORD:
             value = f"**{value}"
+        param_name = etree.SubElement(
+            param, "em", {"class": "autodoc-param autodoc-param-name"}
+        )
+        param_name.text = value
+
+        annotate_parameter = (
+            include_type_annotations and parameter.annotation is not parameter.empty
+        )
+
+        if annotate_parameter:
+            colon = etree.SubElement(param, "span", {"class": "autodoc-punctuation"})
+            colon.text = ": "
+            type_ = etree.SubElement(
+                param, "span", {"class": "autodoc-type-annotation"}
+            )
+            type_.text = render_type_annotation(parameter.annotation)
+        if parameter.default is not parameter.empty:
+            equals = etree.SubElement(param, "span", {"class": "autodoc-punctuation"})
+            equals.text = " = " if annotate_parameter else "="
+            default = etree.SubElement(
+                param, "span", {"class": "autodoc-param-default"}
+            )
+            default.text = f"{parameter.default!r}"
+
         elif parameter.kind is parameter.POSITIONAL_ONLY:
             if render_pos_only_separator:
                 render_pos_only_separator = False
-                params.append("/")
+                pos_seperator = etree.SubElement(
+                    param, "span", {"class": "autodoc-punctuation"}
+                )
+                pos_seperator.text = "/"
         elif parameter.kind is parameter.KEYWORD_ONLY:
             if render_kw_only_separator:
                 render_kw_only_separator = False
-                params.append("*")
-        params.append(value)
+                kw_separator = etree.SubElement(
+                    param, "span", {"class": "autodoc-punctuation"}
+                )
+                kw_separator.text = "*"
+        params.append(param)
 
     return params
 
 
+def render_type_annotation(annotation):
+    """ turn a tree of types into a cleaned string  """
+
+    # gives us string types like Dict[mkautodoc._utils.Foo, List[int]]
+    stringified_annotation = stringify_annotation(annotation)
+
+    # find and clean out module/package stuff (e.g. mkautodoc._utils.Foo -> Foo)
+    return re.sub(r"([^\[^\s]+\.)+", "", stringified_annotation)
```

Comment:
@tomchristie ok, i added this as a cleanup, basically looks for anything after  a `[` or ` ` (whitespace) that ends in a period. it seems to work pretty well, in the tests i found that without this function the sample would have `mocklib.ExampleClass` but after adding this it gets clipped to `ExampleClass`.